anydex/wallet/node/node.py

```diff
@@ -0,0 +1,188 @@
+    A node is an abstraction for a node component in a cryptocurrency network.
+
+    Each cryptocurrency should allow for a user-provided node implementation, as well as
+    a DefaultNode-class implementation.
+    """
+
+    def __init__(self, name: str, host: str, port: int, source: Source,
+                 network: Cryptocurrency, latency: float):
+        self.name = name
+        self.host = host
+        self.port = port
+        self.source = source
+        self.network = network
+        self.latency = latency
+
+    def __repr__(self):
+        return f'{self.name}\n' \
+               f'address: {self.host}:{self.port}\n' \
+               f'source: {self.source}\n' \
+               f'network: {self.network.value}\n' \
+               f'latency: {self.latency}'
+
+
+def create_node(network: Cryptocurrency) -> Node:
+    """
+    Constructs a Node from user-provided parameters if key is present in `config.py`-dictionary.
+    Else: constructs Node picked from set of default nodes provided by AnyDex.
+
+    Raise MissingParameterException if required parameters miss from user Node-config: host, port
+
+    :param network: instance of Cryptocurrency enum
+    :return: Node
+    """
+    config = get_anydex_configuration()
+    params = {'network': network}
+
+    if 'node' in config:
+        _logger.info('Parsing user node config')
+
+        node_config = config['nodes']['node']
+        params['source'] = Source.USER
+        params['name'] = node_config.get('name', '')
+
+        try:
+            params['host'] = node_config['host']
+        except KeyError:
+            raise CannotCreateNodeException('Missing key `host` from node config')
+
+        try:
+            params['port'] = node_config['port']
+        except KeyError:
+            raise CannotCreateNodeException('Missing key `port` from node config')
+
+        params['latency'] = determine_latency(params['host'], params['port'])
+    else:
+        _logger.info('Finding best host from pool of default hosts')
+
+        params['source'] = Source.DEFAULT
+        params['name'] = ''
+
+        default_hosts = read_default_hosts()
+
+        try:
+            network_hosts = default_hosts[network.value]
+        except KeyError:
+            raise CannotCreateNodeException(f'Missing default nodes for {network.value}')
+
+        selected_host, latency = select_best_host(network_hosts)
+        params['host'], params['port'] = selected_host.split(':')
+
+    node = Node(**params)
+    _logger.info(f'Using following node:\n{node}')
+    return node
+
+
+class CannotCreateNodeException(Exception):
+    pass
+
+
+def read_default_hosts():
+    nodes = dict()
+
+    with open('hosts.json') as file:
+        try:
+            nodes = json.loads(file.read())
+        except json.JSONDecodeError as err:
+            _logger.error(f'Default nodes file could not be decoded: {err}')
+
+    return nodes
+
+
+def select_best_host(hosts) -> tuple:
```

Comment:
Could use a short comment to document what _best_ means in this case. I can tell that it is based on latency, but it would be easier for future reference.

Reply:
Will be resolved in the next commit.